src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1283,10 +1283,13 @@ private bool InitializeTaskVectorParameter
         /// </remarks>
         private bool InternalSetTaskParameter(TaskPropertyInfo parameter, IList parameterValue)
         {
-            if (LogTaskInputs && !_taskLoggingContext.LoggingService.OnlyLogCriticalEvents && parameterValue.Count > 0)
+            if (LogTaskInputs &&
+                !_taskLoggingContext.LoggingService.OnlyLogCriticalEvents &&
+                parameterValue.Count > 0 &&
+                parameter.Log)
```

Comment:
For the sake of debuggability , it would be nice to log a comment whenever an item or metadata isn't because these two new knobs are false. And maybe add an escape hatch to ignore them and log everything (maybe reuse the one here [https://github.com/microsoft/msbuild/pull/5210/files#diff-5f4088c7b8512fa52f7faf88f89661f3R122](https://github.com/microsoft/msbuild/pull/5210/files#diff-5f4088c7b8512fa52f7faf88f89661f3R122))

Reply:
We can always add this in the viewer post-build, since we know declaratively which tasks have parameters with logging disabled or constrained. No need to log. E.g. for each Hash task we could easily add an icon or a message: this parameter's item metadata was not logged.

Reply:
And if people are using text logs, it's their fault. (no offense @rainersigwald)